concepts/websockets.textile

```diff
@@ -0,0 +1,591 @@
+
+Example JavaScript code that would have been used to instantiate an @XMLHTTP@ object:
+
+```[js]
+// Note: this code only works in old versions of Internet Explorer
+var xmlhttp = new ActiveXObject("Microsoft.XMLHTTP");
+xmlhttp.open("GET", "/api/example", true);
+xmlhttp.onreadystatechange = function () {
+  if (xmlhttp.readyState == 4) {
+    alert(xmlhttp.responseText);
+  }
+};
+xmlhttp.send(null);
+```
+
+XMLHTTP later became the "XMLHttpRequest":https://xhr.spec.whatwg.org/ defacto standard due to its adoption by other browsers. This was about the time that the term "AJAX" was coined, standing for _Asynchronous JavaScript and XML_. The @JSON@ standard later came along and made everything better, but the _X_ in _AJAX_ (not to mention the _XML_ in _XMLHttpRequest_) never really went away, despite actual XML-formatted data having largely disappeared from standard messaging payloads.
+
+Comparative example of modern use of the @XMLHttpRequest@ object:
+
+```[js]
+const req = new XMLHttpRequest();
+req.addEventListener("load", () => console.log(this.responseText));
+req.open("GET", "/api/example");
+req.send();
+```
+
+As you can see, the modern analog is much the same as the original, albeit with a few additions to make things a little more concise.
+
+h3(#a-world-of-new-possibilities). A world of new possibilities
+
+In any case, XMLHttpRequest was still following the same HTTP request-response model used to retrieve the original HTML document. There was no real notion of allowing a server to proactively contact the user, or to establish any kind of general two-way connection for more sophisticated use cases. All the while though, JavaScript was gaining new features and capabilities as time went on, and browsers were enhancing the document object model (DOM), leading to greater and greater potential with respect to how JavaScript could be used to enrich the user's experience of interacting with a web page.
+
+As the potential for really rich experiences started to become apparent, developers naturally gravitated toward the ideal of client-server applications implemented directly in the browser. Prior to this, the standard paradigm for anything non-trivial was to build a dedicated software application, package it up with an installer, have the user download the installer and then install it natively on their machine. Needless to say, this was quite a barrier to entry for all but the tech-savvy user. Keeping the application up to date with fixes and enhancements was a challenge all of its own. It is easy to understand, then, how alluring it was to be able to build an application that neither required installation before it could be accessed, nor user training and nagging in order to iterate on the software's implementation.
```

Comment:
'toward' -> 'towards'

Reply:
'ideal of client-server applications' -> 'idea of client-server applications'

Reply:
Remove comma after 'accessed'

Reply:
- Toward/towards are both valid - the former is more common in American English.
- "Ideal" is intended as a noun in this case, as distinct from "idea", which has a subtly different meaning.
- This appears to be Oxford comma territory and is thus an issue of style rather than validity